lnwallet/chancloser/chancloser.go

```diff
@@ -419,6 +435,56 @@ func (c *ChanCloser) ShutdownChan() (*lnwire.Shutdown, error) {
 	return shutdownMsg, nil
 }
 
+// MarkChannelClean is used by the caller to notify the ChanCloser that the
+// channel is in a clean state and ClosingSigned negotiation can begin. Any
+// ClosingSigned received before then will be stored in a variable instead of
+// being processed.
+func (c *ChanCloser) MarkChannelClean() ([]lnwire.Message, bool, error) {
+	// Return an error if the state transition is invalid.
+	if c.state != closeFeeNegotiation {
+		return nil, false, fmt.Errorf("channel clean, but not in " +
+			"state closeFeeNegotiation")
+	}
+
+	// Set the fee baseline now that we know the remote's delivery script.
+	c.initFeeBaseline()
+
+	// Before continuing, mark the channel as cooperatively closed with a
+	// nil txn. Even though we haven't negotiated the final txn, this
+	// guarantees that our listchannels rpc will be externally consistent,
+	// and reflect that the channel is being shutdown by the time the
+	// closing request returns.
+	err := c.cfg.Channel.MarkCoopBroadcasted(nil, c.locallyInitiated)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if c.cfg.Channel.IsInitiator() {
+		closeSigned, err := c.proposeCloseSigned(c.idealFeeSat)
```

Comment:
Function seems to do a few things beyond just marking the chan as being "clean". Alternatively, we can send a message to the main shutdown even to process or send a closing signed.